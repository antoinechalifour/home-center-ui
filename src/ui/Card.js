import styled from 'styled-components'

const Card = styled.div`
  background: ${({ theme }) => theme.colors.backgroundInverse};
  color:  ${({ theme }) => theme.colors.textInverse};
  box-shadow: 0 1px 3px rgba(0, 0, 0, .24);
`

export default Card

export const Content = styled.div`
  padding: 12px;
  color: ${({ theme }) => theme.colors.textInverse};
`
