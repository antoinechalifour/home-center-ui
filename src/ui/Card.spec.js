import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import Card, * as card from './Card'

const theme = {
  colors: {
    backgroundInverse: '#f7f7f9',
    textInverse: '#373d3f'
  }
}

describe('ui/Card', () => {
  it('Should render a card', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Card>
          <card.Content>Hello world</card.Content>
        </Card>
      </ThemeProvider>
    )

    expect(tree).toMatchSnapshot()
  })
})
