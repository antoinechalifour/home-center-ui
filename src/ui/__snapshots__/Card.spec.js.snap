// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Card Should render a card 1`] = `
.c0 {
  background: #f7f7f9;
  color: #373d3f;
  box-shadow: 0 1px 3px rgba(0,0,0,.24);
}

.c1 {
  padding: 12px;
  color: #373d3f;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Hello world
  </div>
</div>
`;
